mypy/checkexpr.py

```diff
@@ -1802,10 +1802,22 @@ def check_callable_call(
 
         if (
             callee.is_type_obj()
-            and (len(arg_types) == 1)
+            and len(arg_types) == 1
             and is_equivalent(callee.ret_type, self.named_type("builtins.type"))
         ):
-            callee = callee.copy_modified(ret_type=TypeType.make_normalized(arg_types[0]))
+            proper_arg = get_proper_type(arg_types[0])
+            if isinstance(proper_arg, Instance) and proper_arg.type.is_protocol:
+                callee = callee.copy_modified(
+                    ret_type=UnionType(
+                        [
+                            self.named_type("builtins.type"),
+                            self.named_type("types.ModuleType"),
```

Comment:
This seems wrong: `self.named_type` returns an `Instance`. `type(some_module)` is `types.ModuleType` itself, not its instance. So this should be `type[Any] | type[types.ModuleType] | Any` instead, and at this point `type[types.ModuleType]` becomes redundant.